JS.md

```diff
@@ -0,0 +1,235 @@
+associated with the `Wasm.Module` receiver.
+
+If the `importObject` parameter is not `undefined` and `Type(importObject)` is
+not Object, a `TypeError` is thrown. If the list of 
+[`m.imports`](https://github.com/WebAssembly/spec/blob/master/ml-proto/spec/ast.ml#L218)
+is not empty and `Type(importObject)` is not Object, a `TypeError` is thrown.
+
+Let `imports` by an initially-empty list of JS functions.
+
+For each [`import`](https://github.com/WebAssembly/spec/blob/master/ml-proto/spec/kernel.ml#L133)
+`i` in `m.imports`:
+* Let `v` be the resultant value of performing
+  [`Get`](http://tc39.github.io/ecma262/#sec-get-o-p)(`importObject`, [`i.module_name`](https://github.com/WebAssembly/spec/blob/master/ml-proto/spec/kernel.ml#L137)).
+* If [`i.func_name`](https://github.com/WebAssembly/spec/blob/master/ml-proto/spec/kernel.ml#L138)
+  is not the empty string:
+  * If `Type(v)` is not Object, throw a `TypeError`.
+  * Let `v` instead be the value of performing [`Get`](http://tc39.github.io/ecma262/#sec-get-o-p)(`v`, `i.func_name`)
+* If `IsCallable(v)` is `false`, throw a `TypeError`.
+* Append `v` to `imports`.
+
+Let `instance` be the result of evaluating 
+[`Eval.init`](https://github.com/WebAssembly/spec/blob/master/ml-proto/spec/eval.ml#L314)
+with arguments `m` and `imports`.
+Note: this synchronously executes the [`start`](Modules.md#module-start-function)
+function, if present.
+
+Let `exportObject` be a new, plain empty JS object.
+
+For each [exported function](https://github.com/WebAssembly/spec/blob/master/ml-proto/spec/kernel.ml#L130)
+`f` in `m.exports`:
+* Let `index` be the function index of `f`.
+* If `f` is the first export of `index`, let `v` be a new 
+  [Exported Function](#exported-function-exotic-objects) Exotic Object given `instance` and `index`.
+* Otherwise, let `v` be the previously-created Exported Function Exotic Object.
+* Add a plain data property to `exportObject` whose property name is given by `f` and
+  whose value is `v`.
+
+Return a new `Wasm.Instance` instance containing `instance` where the initial value of
+the [`exports`](#wasminstance-exports-property) data property is
+`exportObject`.
+
+## Exported Function Exotic Objects
+
+Functions exported by WebAssembly modules are reflected in JS via a new kind
+of *Exported Function* 
+[Exotic Object](http://tc39.github.io/ecma262/#sec-built-in-exotic-object-internal-methods-and-slots).
+Like [Bound Function](http://tc39.github.io/ecma262/#sec-bound-function-exotic-objects) Exotic Object,
+Exported Functions do not have the normal function internal slots but instead have:
+ * [[WasmInstance]] : the [`Eval.instance`](https://github.com/WebAssembly/spec/blob/master/ml-proto/spec/eval.ml#L15)
+   containing the exported function
+ * [[WasmFunctionIndex]] : the index of the function inside the module
+
+Exported Functions also have the following data properties:
+* the `length` property is set to the exported function's signature's arity 
```

Comment:
E.g., bound function objects (what you get from `f.bind()`) have a length (reflecting the new arity after binding) and name which is "bound <original function name>".  So yes, I think this is the precedent.